### ISSUE TITLE
[Bug] Monaco autocomplete text is hidden if page CSS defines 'main'

### ISSUE THREAD

**Issue by boourns** [microsoft/monaco-editor#3036]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
/* add to CSS in playground */

.main {
    display: flex;
    flex-direction: column;
    flex: 1;
    overflow: hidden;
}
```

### Actual Behavior

Autocomplete text is missing, shows up blank:

<img width="535" alt="image" src="https://user-images.githubusercontent.com/699550/159926538-0366dcfb-e5fc-4aca-8b4c-7c6f425b2755.png">

### Expected Behavior

autocomplete text should be shown:

<img width="559" alt="image" src="https://user-images.githubusercontent.com/699550/159926657-be7d51c3-83eb-4974-a881-8a806b947560.png">

### Additional Context

`.main` is a very common word, and this bug is tough to debug since the autocomplete menu disappears once I click into the web inspector (on firefox, at least).

I think Monaco should not depend on such a common, un-prefixed CSS class being unused in order to function.  Could we rename Monaco's use  of `.main` to `.monaco-main` or similar?

**Comment by sabhas** [comment]
I would agree with @boourns that it was really hard to debug; we had a similar [issue](https://github.com/sasjs/server/issues/332) in our product and it took me a lot of time to figure it out. I was fortunate that I found this issue in the issue list otherwise I might have wasted some more time.

**Comment by jrieken** [comment]
I understand this but it isn't feasible for to manage a list of prohibited class names. Something like `contents`, `left`, `right` etc pp would all make it onto this list

**Comment by boourns** [comment]
This bug is really easy to accidentally trigger and really hard to debug.  

I would expect a module that gets loaded into the DOM like Monaco to either prefix all of the CSS classes it uses like "monaco-main", make no assumptions about the parent CSS or DOM state, or at least document that one cannot use "main", an incredibly common word.
